common/remote/sockfile.cpp

```diff
@@ -4360,10 +4360,10 @@ class CRemoteIndexCountActivity : public CRemoteIndexBaseActivity
         inMeta.setown(getTypeInfoOutputMetaData(config, "input", false));
 
         // create a { unsigned8 } output meta for the count
-        const RtlIntTypeInfo countFieldType(type_int, 8);
-        const RtlFieldStrInfo countField("count", nullptr, &countFieldType);
-        const RtlFieldInfo * const fields[2] = { &countField, nullptr };
-        const RtlRecordTypeInfo record = RtlRecordTypeInfo(type_record, 2, fields);
+        static const RtlIntTypeInfo countFieldType(type_unsigned|type_int, 8);
```

Comment:
I think much better to move them outside the function.  Otherwise they are probably created when the function is first called (and need protection from multiple threads).

Reply:
In looking at this discovered that thread safe in c++11.

> section 6.7:
> 
> such a variable is initialized the first time control passes through its declaration; such a variable
> is considered initialized upon the completion of its initialization. [...] If control enters the
> declaration concurrently while the variable is being initialized, the concurrent execution shall
> wait for completion of the initialization.


I thought thread safe before, but they didn't used to be before c++11.